app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeight.java

```diff
@@ -0,0 +1,99 @@
+package de.westnordost.streetcomplete.quests.max_height;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import de.westnordost.osmapi.map.data.BoundingBox;
+import de.westnordost.osmapi.map.data.Element;
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.osm.Countries;
+import de.westnordost.streetcomplete.data.osm.OsmElementQuestType;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.data.osm.tql.OverpassQLUtil;
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment;
+
+public class AddMaxHeight implements OsmElementQuestType
+{
+
+	private final OverpassMapDataDao overpassServer;
+
+	@Inject public AddMaxHeight(OverpassMapDataDao overpassServer)
+	{
+		this.overpassServer = overpassServer;
+	}
+
+	/*the query does not exclude elements which are not accessible by the public,
+	because a maxheight sign will very probably be visible at the entrance or beginning*/
+	private static final String QUERY_RESTRICTIONS = "['maxheight'!~'.']['maxheight:physical'!~'.']";
```

Comment:
You can use `!tag` instead of `"tag"!~'.'` here : e.g. `[!maxheight]`, or `[!"maxheight:physical"]`